src/ui-components/rename-dialog.tsx

```diff
@@ -0,0 +1,65 @@
+import { showDialog, Dialog } from '@jupyterlab/apputils';
+import { ReactWidget } from '@jupyterlab/apputils';
+import React from 'react';
+
+/**
+ * The value returned by the Rename dialog.
+ */
+export interface IRenameDialogValue {
+  newName: string;
+}
+
+/**
+ * The body of the Rename dialog, containing a single input field.
+ * The current name is passed in the constructor.
+ */
+class RenameDialogBody extends ReactWidget {
+  private _value: string;
+
+  constructor(private _currentName: string) {
+    super();
+    this._value = _currentName;
+    this.addClass('je-RenameDialog');
+  }
+
+  getValue(): IRenameDialogValue {
+    return { newName: (this._value ?? '').trim() };
+  }
+
+  protected render(): React.ReactElement {
+    return (
+      <div className="je-RenameDialog-body">
+        <label htmlFor="je-rename-input" className="je-RenameDialog-label">
+          New name
+        </label>
+        <input
+          id="je-rename-input"
+          type="text"
+          defaultValue={this._currentName}
+          onChange={e => {
+            this._value = (e.target as HTMLInputElement).value;
+          }}
+          autoFocus
```

Comment:
Th autofocus does not work because the dialog itself focuses on the cancel button.

Instead you need to pass `focusNodeSelector: 'input'` to `showDialog`.

If we are to stick to the dialog for now I would suggest using the same approach as core lab does of making the selection range not include the file extension by default:

```ts
const ext = PathExt.extname(oldPath);
const value = (this.inputNode.value = PathExt.basename(oldPath));
this.inputNode.setSelectionRange(0, value.length - ext.length);
```

https://github.com/jupyterlab/jupyterlab/blob/c836a03e8ee2ecc17da83c904d2fe93470dfa8fa/packages/docmanager/src/dialogs.ts#L273

Reply:
> using the same approach as core lab does of making the selection range not include the file extension by default

Thanks! I actually did try to do this, but it didn't work because of the focus on the cancel button. On it!